### PR TITLE
refactor code structure

### PR DESCRIPTION
**Pull request by frjcomp**

### CODE REVIEW

**Comment by frjcomp** [comment]
@copilot i want you to analyze how we can integrate  /workspaces/pipeleak/src/pipeleak/pkg/gitlab/scan with /workspaces/pipeleak/src/pipeleak/pkg/scan and in /workspaces/pipeleak/src/pipeleak/pkg/scanner like the other scan commands do. Then refactor gitlab scan to remove duplicate logic.

**Comment by frjcomp** [comment]
@copilot You are Copilot working on CompassSecurity/pipeleak (Go). Your goal: find packages under src/pipeleak that lack meaningful unit tests and add only meaningful *_test.go files next to those packages.

Do this concisely:

Scope: src/pipeleak only. Do NOT edit e2e tests or files outside src/pipeleak.
Test criteria: add tests that exercise real logic — error paths, edge cases, parsing, filtering, archive handling, API translation. Do NOT add trivial getters/setters tests.
Conventions: Go table-driven tests, use testify/assert, t.Parallel() where safe, t.TempDir() for IO, mock external clients via interfaces (fakes in _test.go), deterministic and fast.
Implementation: for each package produce (1) a short list of target functions/methods and why a test is needed, then (2) add *_test.go files covering normal, edge, and error cases. Use minimal production-code changes only if absolutely needed for testability (add tiny adapter interfaces).
Verification: run golangci-lint and go test ./... (excluding tests/e2e). Include test results in PR/commit message.
Deliverable: added test files, one-line rationale per file, and commands+results.

File: src/pipeleak/cmd/internal/flags/common.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+	"github.com/spf13/cobra"
+)
+
+// AddCommonScanFlags adds the standard scanning flags that are common across all platforms.
+// These flags control scanning behavior, thread count, verification, and filtering.
+func AddCommonScanFlags(cmd *cobra.Command, opts *config.CommonScanOptions, maxArtifactSize *string) {
+	cmd.Flags().IntVarP(&opts.MaxScanGoRoutines, "threads", "", 4, "Number of concurrent threads for scanning")
+	cmd.Flags().BoolVarP(&opts.TruffleHogVerification, "truffle-hog-verification", "", true,
+		"Enable TruffleHog credential verification to actively test found credentials and only report verified ones (enabled by default, disable with --truffle-hog-verification=false)")
+	cmd.Flags().BoolVarP(&opts.Artifacts, "artifacts", "a", false, "Scan workflow artifacts")
```
Comment (frjcomp) [inline src/pipeleak/cmd/internal/flags/common.go]: rename it to "Scan Artifacts" as workflow is platform specific

File: src/pipeleak/cmd/internal/flags/common.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	cmd.Flags().StringVarP(maxArtifactSize, "max-artifact-size", "", "500Mb",
+		"Maximum artifact size to scan. Larger files are skipped. Format: https://pkg.go.dev/github.com/docker/go-units#FromHumanSize")
+	cmd.Flags().StringSliceVarP(&opts.ConfidenceFilter, "confidence", "", []string{},
+		"Filter for confidence level, separate by comma if multiple. See readme for more info.")
+	cmd.Flags().BoolVarP(&opts.Owned, "owned", "o", false, "Scan only user owned repositories")
```
Comment (frjcomp) [inline src/pipeleak/cmd/internal/flags/common.go]: is owned actually implemented everywhere you use this helper?

File: src/pipeleak/cmd/gitea/scan/scan.go [hunk 3]
```diff
@@ -66,47 +65,41 @@ pipeleak gitea scan --token gitea_token_xxxxx --gitea https://gitea.example.com
...
 	scanCmd.Flags().Int64VarP(&scanOptions.StartRunID, "start-run-id", "", 0, "Start scanning from a specific run ID (only valid with --repository flag, 0 = start from latest)")
-	scanCmd.Flags().StringSliceVarP(&scanOptions.ConfidenceFilter, "confidence", "", []string{}, "Filter for confidence level, separate by comma if multiple. See documentation for more info.")
-	scanCmd.PersistentFlags().IntVarP(&scanOptions.MaxScanGoRoutines, "threads", "", 4, "Nr of threads used to scan")
-	scanCmd.PersistentFlags().BoolVarP(&scanOptions.TruffleHogVerification, "truffle-hog-verification", "", true, "Enable TruffleHog credential verification to actively test found credentials and only report verified ones (enabled by default, disable with --truffle-hog-verification=false)")
 
 	return scanCmd
 }
 
 func Scan(cmd *cobra.Command, args []string) {
+	// Get token and gitea from parent persistent flags
```
Comment (frjcomp) [inline src/pipeleak/cmd/gitea/scan/scan.go]: remvove all those comments in this commit, theyre not meaningful

**Comment by frjcomp** [comment]
@copilot the codespace containerconfig is not working.

update the config to use the github standard golang image and remove the user definition if not absouletely necessray

```
the setup of the codespace failed 
=================================================================================
2025-11-25 11:30:11.469Z: Creating container...
2025-11-25 11:30:11.572Z: $ devcontainer up --id-label Type=codespaces --workspace-folder /var/lib/docker/codespacemount/workspace/pipeleak --mount type=bind,source=/.codespaces/agent/mount/cache,target=/vscode --user-data-folder /var/lib/docker/codespacemount/.persistedshare --container-data-folder .vscode-remote/data/Machine --container-system-data-folder /var/vscode-remote --log-level trace --log-format json --update-remote-user-uid-default never --mount-workspace-git-root false --omit-config-remote-env-from-metadata --skip-non-blocking-commands --skip-post-create --expect-existing-container --config "/var/lib/docker/codespacemount/workspace/pipeleak/.devcontainer/devcontainer.json" --override-config /root/.codespaces/shared/merged_devcontainer.json --default-user-env-probe loginInteractiveShell --container-session-data-folder /workspaces/.codespaces/.persistedshare/devcontainers-cli/cache --secrets-file /root/.codespaces/shared/user-secrets-envs.json
2025-11-25 11:30:11.713Z: @devcontainers/cli 0.80.0. Node.js v18.20.6. linux 6.8.0-1030-azure x64.
2025-11-25 11:30:11.899Z: $ docker start 7ab54ebc26909230ae0781fa5e5078355c7ac7fa78c53ba175e3b1493f11b9a9
2025-11-25 11:30:12.269Z: 7ab54ebc26909230ae0781fa5e5078355c7ac7fa78c53ba175e3b1493f11b9a9

2025-11-25 11:30:12.272Z: Stop: Run: docker start 7ab54ebc26909230ae0781fa5e5078355c7ac7fa78c53ba175e3b1493f11b9a9
2025-11-25 11:30:12.348Z: Shell server terminated (code: 126, signal: null)
2025-11-25 11:30:12.349Z: {"outcome":"error","message":"An error occurred setting up the container.","description":"An error occurred setting up the container.","containerId":"7ab54ebc26909230ae0781fa5e5078355c7ac7fa78c53ba175e3b1493f11b9a9"}
2025-11-25 11:30:12.349Z: unable to find user vscode: no matching entries in passwd file
2025-11-25 11:30:12.351Z: Error: An error occurred setting up the container.
2025-11-25 11:30:12.352Z:     at w6 (/.codespaces/agent/bin/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:467:1253)
2025-11-25 11:30:12.352Z:     at ax (/.codespaces/agent/bin/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:467:997)
2025-11-25 11:30:12.352Z:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2025-11-25 11:30:12.353Z:     at async Y6 (/.codespaces/agent/bin/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:484:3842)
2025-11-25 11:30:12.354Z:     at async BC (/.codespaces/agent/bin/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:484:4957)
2025-11-25 11:30:12.354Z:     at async p7 (/.codespaces/agent/bin/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:665:202)
2025-11-25 11:30:12.355Z:     at async d7 (/.codespaces/agent/bin/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:664:14804)
2025-11-25 11:30:12.355Z: devcontainer process exited with exit code 1
2025-11-25 11:30:12.355Z:     at async /.codespaces/agent/bin/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:484:1188

====================================== ERROR ====================================
2025-11-25 11:30:12.361Z: Failed to create container.
=================================================================================
2025-11-25 11:30:12.361Z: Error: An error occurred setting up the container.
2025-11-25 11:30:12.364Z: Error code: 1302 (UnifiedContainersErrorFatalCreatingContainer)`
````
